src/components/HistoricalForecastWidget/HistoricalForecastWidget.tsx

```diff
@@ -0,0 +1,123 @@
+import { useEffect, useState } from 'react';
+import { format, fromUnixTime } from 'date-fns';
+import groupBy from 'lodash/groupBy';
+import pick from 'lodash/pick';
+import CircularProgress from '@mui/material/CircularProgress';
+import Tooltip from '@mui/material/Tooltip';
+import { useSnackbar } from 'notistack';
+import { FormValuesType } from 'components/HistoricalForecast/validation';
+import TemperatureUnits from 'constants/temperatureUnits';
+import ForecastBody from 'types/forecastBody';
+import findCountryNameByCode from 'utils/findCountryNameByCode';
+import FORECAST from './forecast';
+import WEEK_DAYS from './weekDays';
+
+type Forecast = Array<[string, [ForecastBody]]>;
```

Comment:
1. Should be a tuple type.
2. I think that the main problem is that in `WeatherWidget` we need **an array of objects** for specific date, in this module you need a **single object** for date; so logic on lines 38-43 is redundant